static/scss/components/filter-bar.scss

```diff
@@ -215,7 +215,8 @@
     top: calc(100% + #{$spacing-md});
     background-color: $color-white;
     box-shadow: $box-shadow-sm;
-    z-index: 1;
+    // #fx-private-relay/issues/1605: Z-index is 2 to appear over the Promo Blocking UI form
```

Comment:
Nit: Ideally there's also a reference to the selector that applies the z-index of 1, so that can be checked if it starts overlapping this one again.